File: Exercise 1 Solution/app/src/main/java/com/example/exercise1task/MainActivity.java
package com.example.exercise1task;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

/**
* <h2>SOLUTION for exercise 1 <em>(10 minutes)</em>:</h1>
 * <ul>
 * <li>a) Add an <code>EditText</code> to your layout.</li>
 * <li>b) Add a <code>Button</code> that, when clicked, will make your
 * app log the text input givená by the user using <code>Log.v()</code>.</li>
 * <li>c) Show the result in a <code>TextView</code>.</li>
 * </ul>
 * */
public class MainActivity extends AppCompatActivity {

    TextView textView;
    EditText editText;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        editText = findViewById(R.id.editText);
        button = findViewById(R.id.button);
        textView = findViewById(R.id.textView);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {



                String text = editText.getText().toString();


                Log.v("GDG", "Text from TextEdit: " + text);
                textView.setText(text);
            }
        });
    }

}

File: Exercise 3 Solution/app/src/main/java/com/example/exercise3task/MainActivity.kt
package com.example.exercise3task

import android.os.Bundle
import android.util.Log
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Column
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier

/**
 * Exercise 2 (10 minutes):
 *
 * Recreate the UI and functionality of your app from the previous exercise,
 * using Kotlin + Compose.
 * */
class MainActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            var text by remember { mutableStateOf("") }
            var pageChanged by remember { mutableStateOf(false) }

            if (!pageChanged) {
                Home(value = text,
                    onValueChange = {
                        text = it
                    },
                    onClick = {
                        pageChanged = true
                        Log.v("GDG", "Button clicked")
                    })
            } else {
                Page2(value = text)
            }
        }
    }


}

@Composable
fun Home(modifier: Modifier = Modifier,
         value: String,
         onValueChange: (String) -> Unit,
         onClick: () -> Unit) {

    Column(modifier = modifier) {
        TextField(modifier = modifier, value = value, onValueChange = onValueChange)
        Button(modifier = modifier, onClick = onClick) {
            Text("Submit")
        }
    }

}

@Composable
fun Page2(value: String) {
        Text(value)
}